test/annotate_attribute.ll

```diff
@@ -0,0 +1,141 @@
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+; CHECK-SPIRV: Decorate {{[0-9]+}} UserSemantic "42"
+; CHECK-SPIRV: Decorate {{[0-9]+}} UserSemantic "bar"
+; CHECK-SPIRV: Decorate {{[0-9]+}} UserSemantic "{FOO}"
+; CHECK-SPIRV: MemberDecorate {{[0-9]+}} 1 UserSemantic "128"
+; CHECK-SPIRV: MemberDecorate {{[0-9]+}} 0 UserSemantic "{baz}"
+
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir64-unknown-linux"
+
+%class.anon = type { i8 }
+%struct.bar = type { i32, i8 }
+
+; CHECK-LLVM: [[STR1:@[a-zA-Z0-9_.]+]] = {{.*}}42
+; CHECK-LLVM: [[STR2:@[a-zA-Z0-9_.]+]] = {{.*}}{FOO}
+; CHECK-LLVM: [[STR3:@[a-zA-Z0-9_.]+]] = {{.*}}bar
+; CHECK-LLVM: [[STR4:@[a-zA-Z0-9_.]+]] = {{.*}}{baz}
+; CHECK-LLVM: [[STR5:@[a-zA-Z0-9_.]+]] = {{.*}}128
+@.str = private unnamed_addr constant [3 x i8] c"42\00", section "llvm.metadata"
+@.str.1 = private unnamed_addr constant [23 x i8] c"annotate_attribute.cpp\00", section "llvm.metadata"
+@.str.2 = private unnamed_addr constant [6 x i8] c"{FOO}\00", section "llvm.metadata"
+@.str.3 = private unnamed_addr constant [4 x i8] c"bar\00", section "llvm.metadata"
+@.str.4 = private unnamed_addr constant [6 x i8] c"{baz}\00", section "llvm.metadata"
+@.str.5 = private unnamed_addr constant [4 x i8] c"128\00", section "llvm.metadata"
+
+; Function Attrs: nounwind
+define spir_kernel void @_ZTSZ4mainE15kernel_function() #0 !kernel_arg_addr_space !4 !kernel_arg_access_qual !4 !kernel_arg_type !4 !kernel_arg_base_type !4 !kernel_arg_type_qual !4 {
+entry:
+  %0 = alloca %class.anon, align 1
+  %1 = bitcast %class.anon* %0 to i8*
+  call void @llvm.lifetime.start.p0i8(i64 1, i8* %1) #4
+  call spir_func void @"_ZZ4mainENK3$_0clEv"(%class.anon* %0)
+  %2 = bitcast %class.anon* %0 to i8*
+  call void @llvm.lifetime.end.p0i8(i64 1, i8* %2) #4
+  ret void
+}
+
+; Function Attrs: argmemonly nounwind
+declare void @llvm.lifetime.start.p0i8(i64, i8* nocapture) #1
+
+; Function Attrs: inlinehint nounwind
+define internal spir_func void @"_ZZ4mainENK3$_0clEv"(%class.anon* %this) #2 align 2 {
+entry:
+  %this.addr = alloca %class.anon*, align 8
+  store %class.anon* %this, %class.anon** %this.addr, align 8, !tbaa !5
+  %this1 = load %class.anon*, %class.anon** %this.addr, align 8
+  call spir_func void @_Z3foov()
+  call spir_func void @_Z3bazv()
+  ret void
+}
+
+; Function Attrs: argmemonly nounwind
+declare void @llvm.lifetime.end.p0i8(i64, i8* nocapture) #1
+
+; Function Attrs: nounwind
+define spir_func void @_Z3foov() #3 {
+entry:
+  %var_one = alloca i32, align 4
+  %var_two = alloca i32, align 4
+  %var_three = alloca i8, align 1
+  %0 = bitcast i32* %var_one to i8*
+  call void @llvm.lifetime.start.p0i8(i64 4, i8* %0) #4
+  %var_one1 = bitcast i32* %var_one to i8*
+  ; CHECK-LLVM: call void @llvm.var.annotation(i8* %[[VAR1:[a-zA-Z0-9_]+]], i8* getelementptr inbounds ([3 x i8], [3 x i8]* [[STR1]], i32 0, i32 0), i8* undef, i32 undef)
```

Comment:
`STR1` is equal to `42`, while  `@.str.1` is equal to `annotate_attribute.cpp\00`, how does this test work?

Reply:
Didn't get your concern. The test is similar to existing tests on attributes translation, e.g. `IntelFPGAMemoryAttributes.ll`.

Reply:
I'm just looking at llvm ir:

```
call void @llvm.var.annotation( ... @.str.1, ...)
; where @.str.1 is "annotate_attribute.cpp\00"
```
And your check:
```
call void @llvm.var.annotation( ... [[STR1]], ...)
; where STR1 is "42"
```

I wonder how does it work if you are checking for different annotation string

Reply:
The test checks only first two arguments of var/ptr.annotation intrinsic. 

The first argument is a pointer to a value, the second is a pointer to a global string with annotation.
The third argument `@.str.1` with source file name and the fourth argument with line number are not translated from SPIR-V to LLVM-IR, so they are not checked.

Numeration of `STR1, STR2, ...` is related only to global annotation strings, they are `'42', '{FOO}', ..`.